--- origsrc/libvirt-3.1.0/src/hyperv/hyperv_wmi.c	2015-01-23 05:46:24.727442300 -0600
+++ src/libvirt-3.1.0/src/hyperv/hyperv_wmi.c	2017-03-16 20:20:57.116642400 -0500
@@ -40,7 +40,7 @@
     "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/cimv2/*"
 
 #define ROOT_VIRTUALIZATION \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/*"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/*"
 
 #define VIR_FROM_THIS VIR_FROM_HYPERV
 
--- origsrc/libvirt-3.1.0/src/hyperv/hyperv_wmi_classes.generated.h	2016-09-27 14:47:30.364005200 -0500
+++ src/libvirt-3.1.0/src/hyperv/hyperv_wmi_classes.generated.h	2017-03-16 20:23:56.765569900 -0500
@@ -66,7 +66,7 @@ struct _CIM_DataFile {
  */
 
 #define MSVM_ALLOCATIONCAPABILITIES_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_AllocationCapabilities"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_AllocationCapabilities"
 
 #define MSVM_ALLOCATIONCAPABILITIES_CLASSNAME \
     "Msvm_AllocationCapabilities"
@@ -103,7 +103,7 @@ struct _Msvm_AllocationCapabilities {
  */
 
 #define MSVM_COMPUTERSYSTEM_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_ComputerSystem"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_ComputerSystem"
 
 #define MSVM_COMPUTERSYSTEM_CLASSNAME \
     "Msvm_ComputerSystem"
@@ -158,7 +158,7 @@ struct _Msvm_ComputerSystem {
  */
 
 #define MSVM_CONCRETEJOB_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_ConcreteJob"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_ConcreteJob"
 
 #define MSVM_CONCRETEJOB_CLASSNAME \
     "Msvm_ConcreteJob"
@@ -220,7 +220,7 @@ struct _Msvm_ConcreteJob {
  */
 
 #define MSVM_MEMORYSETTINGDATA_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_MemorySettingData"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_MemorySettingData"
 
 #define MSVM_MEMORYSETTINGDATA_CLASSNAME \
     "Msvm_MemorySettingData"
@@ -271,7 +271,7 @@ struct _Msvm_MemorySettingData {
  */
 
 #define MSVM_PROCESSORSETTINGDATA_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_ProcessorSettingData"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_ProcessorSettingData"
 
 #define MSVM_PROCESSORSETTINGDATA_CLASSNAME \
     "Msvm_ProcessorSettingData"
@@ -326,7 +326,7 @@ struct _Msvm_ProcessorSettingData {
  */
 
 #define MSVM_RESOURCEALLOCATIONSETTINGDATA_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_ResourceAllocationSettingData"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_ResourceAllocationSettingData"
 
 #define MSVM_RESOURCEALLOCATIONSETTINGDATA_CLASSNAME \
     "Msvm_ResourceAllocationSettingData"
@@ -374,7 +374,7 @@ struct _Msvm_ResourceAllocationSettingDa
  */
 
 #define MSVM_SWITCHPORT_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_SwitchPort"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_SwitchPort"
 
 #define MSVM_SWITCHPORT_CLASSNAME \
     "Msvm_SwitchPort"
@@ -431,7 +431,7 @@ struct _Msvm_SwitchPort {
  */
 
 #define MSVM_SYNTHETICETHERNETPORTSETTINGDATA_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_SyntheticEthernetPortSettingData"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_SyntheticEthernetPortSettingData"
 
 #define MSVM_SYNTHETICETHERNETPORTSETTINGDATA_CLASSNAME \
     "Msvm_SyntheticEthernetPortSettingData"
@@ -480,7 +480,7 @@ struct _Msvm_SyntheticEthernetPortSettin
  */
 
 #define MSVM_VIRTUALHARDDISKSETTINGDATA_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_VirtualHardDiskSettingData"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_VirtualHardDiskSettingData"
 
 #define MSVM_VIRTUALHARDDISKSETTINGDATA_CLASSNAME \
     "Msvm_VirtualHardDiskSettingData"
@@ -519,7 +519,7 @@ struct _Msvm_VirtualHardDiskSettingData
  */
 
 #define MSVM_VIRTUALSWITCH_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_VirtualSwitch"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_VirtualSwitch"
 
 #define MSVM_VIRTUALSWITCH_CLASSNAME \
     "Msvm_VirtualSwitch"
@@ -574,7 +574,7 @@ struct _Msvm_VirtualSwitch {
  */
 
 #define MSVM_VIRTUALSWITCHMANAGEMENTSERVICE_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_VirtualSwitchManagementService"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_VirtualSwitchManagementService"
 
 #define MSVM_VIRTUALSWITCHMANAGEMENTSERVICE_CLASSNAME \
     "Msvm_VirtualSwitchManagementService"
@@ -621,7 +621,7 @@ struct _Msvm_VirtualSwitchManagementServ
  */
 
 #define MSVM_VIRTUALSYSTEMGLOBALSETTINGDATA_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_VirtualSystemGlobalSettingData"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_VirtualSystemGlobalSettingData"
 
 #define MSVM_VIRTUALSYSTEMGLOBALSETTINGDATA_CLASSNAME \
     "Msvm_VirtualSystemGlobalSettingData"
@@ -668,7 +668,7 @@ struct _Msvm_VirtualSystemGlobalSettingD
  */
 
 #define MSVM_VIRTUALSYSTEMMANAGEMENTSERVICE_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_VirtualSystemManagementService"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_VirtualSystemManagementService"
 
 #define MSVM_VIRTUALSYSTEMMANAGEMENTSERVICE_CLASSNAME \
     "Msvm_VirtualSystemManagementService"
@@ -715,7 +715,7 @@ struct _Msvm_VirtualSystemManagementServ
  */
 
 #define MSVM_VIRTUALSYSTEMSETTINGDATA_RESOURCE_URI \
-    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/Msvm_VirtualSystemSettingData"
+    "http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/Msvm_VirtualSystemSettingData"
 
 #define MSVM_VIRTUALSYSTEMSETTINGDATA_CLASSNAME \
     "Msvm_VirtualSystemSettingData"
--- origsrc/libvirt-3.1.0/src/hyperv/hyperv_wmi_generator.py	2016-09-27 14:44:46.091815200 -0500
+++ src/libvirt-3.1.0/src/hyperv/hyperv_wmi_generator.py	2017-03-16 20:20:57.132265300 -0500
@@ -71,7 +71,7 @@ class Class:
         if self.name.startswith("Win32_") or self.name.startswith("CIM_"):
             header += "    \"http://schemas.microsoft.com/wbem/wsman/1/wmi/root/cimv2/%s\"\n" % self.name
         else:
-            header += "    \"http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/%s\"\n" % self.name
+            header += "    \"http://schemas.microsoft.com/wbem/wsman/1/wmi/root/virtualization/v2/%s\"\n" % self.name
 
         header += "\n"
         header += "#define %s_CLASSNAME \\\n" % name_upper
